pkg/event/event.go

```diff
@@ -205,9 +277,17 @@ func (e Event) InngestMetadata() (*InngestMetadata, error) {
 	}
 }
 
-func NewOSSTrackedEvent(e Event) TrackedEvent {
+func NewOSSTrackedEvent(e Event, seed *SeededID) TrackedEvent {
 	// Never use e.ID as the internal ID, since it's specified by the sender
 	internalID := ulid.MustNew(ulid.Now(), rand.Reader)
+
+	if seed != nil {
+		newInternalID, err := seed.ToULID()
+		if err == nil {
+			internalID = newInternalID
+		}
+	}
```

Comment:
Can we add a comment here to note the potential downstream effect of having duplicate IDs in the store?

As you say, it shouldn't actually affect downstream queries, but will be useful if someone finds two events and is confused.